### ISSUE TITLE
Assigning aux variable to a kernel variable does not cause an error

### ISSUE THREAD

**Issue by YaqiWang** [idaholab/moose#11021]
## Rationale
We noticed that assigning an aux variable to a kernel's `variable` parameter does not error out.

## Description
This could potentially create subtle bugs when constructing models. This can be easily reproduced with
```
diff --git a/test/tests/kernels/2d_diffusion/2d_diffusion_test.i b/test/tests/kernels/2d_diffusion/2d_diffusion_test.i
index cdc4fc2e94..d7777c0d20 100644
--- a/test/tests/kernels/2d_diffusion/2d_diffusion_test.i
+++ b/test/tests/kernels/2d_diffusion/2d_diffusion_test.i
@@ -20,13 +20,20 @@
   [../]
 []
 
-[Kernels]
-  active = 'diff'
+[AuxVariables]
+  [./v]
+  [../]
+[]
 
+[Kernels]
   [./diff]
     type = Diffusion
     variable = u
   [../]
+  [./rea]
+    type = Reaction
+    variable = v
+  [../]
 []
 
 [BCs]
@@ -53,6 +60,13 @@
   solve_type = 'NEWTON'
 []
 
+[Postprocessors]
+  [./unorm]
+    type = ElementL2Norm
+    variable = u
+  [../]
+[]
+
 [Outputs]
   file_base = out
   exodus = true
``` 
Note the converged `u` norms are the same but the convergence history is changed.

## Impact
Robust interface.

**Comment by YaqiWang** [comment]
You may want to double check boundary conditions, dg kernels, constraints, etc.

**Comment by permcody** [comment]
This is a pretty silly problem and I'm surprised we don't have a test nor the proper checks in place to make sure this doesn't happen. 

@idaholab/moose-team - Can somebody take a look at this and get this issue patched?

**Comment by jwpeterson** [comment]
@permcody I talked with @andrsd and I am planning to add checks for this in `FEProblemBase::addKernel()`, `addBoundaryCondition()`, etc.

**Comment by andrsd** [comment]
> I'm surprised we don't have a test nor the proper checks in place to make sure this doesn't happen.

We used to have a check. The code was doing `_var(_sys.getVariable(name))`, where `_sys` was either nonlinear system or auxiliary system based on the MOOSE system (i.e. kernels, BCs, Auxs, etc.) It would throw an error if you asked for a wrong variable name within the system. However, this went away with the vector FE pull request. Now we do `problem.getVariable(name)` inside `MooseVariableInterface` where we have no idea which MOOSE system we are in.  And since there was no test, this change went through without noticing...

**Comment by jwpeterson** [comment]
Fixed by #11039.
